drivers/ethernet/phy/phy_motorcomm_yt8521.c

```diff
@@ -0,0 +1,696 @@
+	uint32_t cnt = 1000;
+	uint32_t val = 0;
+
+	do {
+		if (mc_ytphy_read(dev, MII_PHYID2R, &val) < 0) {
+			k_msleep(1);
+		}
+	} while (val != PHY_ID_YT8521 && --cnt > 0);
+
+	if (cnt == 0U) {
+		LOG_ERR("PHY (%d) timeout to get PHY ID", config->phy_addr);
+		return -EIO;
+	}
+
+	if (val != PHY_ID_YT8521) {
+		LOG_ERR("PHY (%d) can't get PHY ID, value:0x%X", config->phy_addr, val);
+		return -EIO;
+	}
+
+	if (phy_id) {
+		*phy_id = val;
+	} else {
+		LOG_INF("PHY (%d) ID:0x%X", config->phy_addr, val);
+	}
+
+	return 0;
+}
+
+static int mc_ytphy_init(const struct device *dev)
+{
+	const struct mc_ytphy_config *const config = dev->config;
+	struct mc_ytphy_data *const data = dev->data;
+	int ret;
+
+	k_sem_init(&data->sem, 1, 1);
+
+	mdio_bus_enable(config->mdio);
+
+	data->state.is_up = false;
+	data->dev = dev;
+	data->cb = NULL;
+
+	if (mc_ytphy_get_id(dev, NULL)) {
+		return -EIO;
+	}
+
+	/* set default reg space */
+	mc_ytphy_write_ext(dev, YT8521_REG_SPACE_SELECT_REG, YT8521_RSSR_UTP_SPACE);
+
+	mc_ytphy_modify_ext(dev, YTPHY_SYNCE_CFG_REG, YT8521_SCR_SYNCE_ENABLE, 0);
+
+	/* Reset PHY */
+	ret = mc_ytphy_soft_reset(dev);
+	if (ret) {
+		return -EIO;
+	}
+
+	/* Enable clock delay */
+	mc_ytphy_cfg_clock_delay(dev);
+
+	mc_ytphy_resume(dev);
+
+	return 0;
+}
+
+#if ANY_SPECIFIED_LINK
+static int mc_ytphy_initialize_specified_link(const struct device *dev)
+{
+	const struct mc_ytphy_config *const cfg = dev->config;
+	struct mc_ytphy_data *const data = dev->data;
+	uint16_t mask, bmcr_val = 0;
+	int ret;
+
+	const static int speed_to_phy_link_speed[] = {
+		LINK_HALF_10BASE,  LINK_FULL_10BASE,   LINK_HALF_100BASE,
+		LINK_FULL_100BASE, LINK_HALF_1000BASE, LINK_FULL_1000BASE,
+	};
+
+	ret = mc_ytphy_init(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	data->state.speed = speed_to_phy_link_speed[cfg->specified_speed];
+
+	bmcr_val = mii_bmcr_encode_specified(data->state.speed);
+	LOG_INF("PHY (%d) Specified link speed %s Mb, %s duplex", cfg->phy_addr,
+		PHY_LINK_IS_SPEED_1000M(data->state.speed)
+			? "1000"
+			: (PHY_LINK_IS_SPEED_100M(data->state.speed) ? "100" : "10"),
+		PHY_LINK_IS_FULL_DUPLEX(data->state.speed) ? "full" : "half");
+
+	mask = 0xFFFF & ~(MII_BMCR_LOOPBACK | MII_BMCR_ISOLATE | MII_BMCR_POWER_DOWN);
+	mc_ytphy_modify(dev, MII_BMCR, mask, bmcr_val);
+
+	data->state.is_up = true;
+
+	return 0;
+}
+#endif /* ANY_SPECIFIED_LINK */
+
+#if ANY_DYNAMIC_LINK
+static int mc_ytphy_initialize_dynamic_link(const struct device *dev)
+{
+	struct mc_ytphy_data *const data = dev->data;
+	uint32_t phy_feature;
+	int ret = 0;
+
+	ret = mc_ytphy_init(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	data->state.is_up = false;
+
+	k_work_init_delayable(&data->monitor_work, monitor_work_handler);
+
+	/* Auto negotiation, advertise all supported speeds */
+	phy_feature = LINK_FULL_100BASE | LINK_HALF_100BASE | LINK_FULL_10BASE
+		| LINK_HALF_10BASE | LINK_FULL_1000BASE;
+	mc_ytphy_cfg_link(dev, phy_feature, 0);
```

Comment:
use the ``default-speeds`` from the dt, it was introduced, so users could change the advertised speeds for autoneg from the dt.

Reply:
this has to be addressed @JiafeiPan

Reply:
Updated to add default_speeds support, it will advertise all supported speeds in default_speeds not provided in dt.